source/AI.cpp

```diff
@@ -3052,13 +3038,13 @@ void AI::AutoFire(const Ship &ship, FireCommand &command, bool secondary) const
 	// Find the longest range of any of your non-homing weapons. Homing weapons
 	// that don't consume ammo may also fire in non-homing mode.
 	double maxRange = 0.;
-	for(const Hardpoint &weapon : ship.Weapons())
-		if(weapon.IsReady()
-				&& !(!currentTarget && weapon.IsHoming() && weapon.GetOutfit()->Ammo())
-				&& !(!secondary && weapon.GetOutfit()->Icon())
-				&& !(beFrugal && weapon.GetOutfit()->Ammo())
-				&& !(isWaitingToJump && weapon.GetOutfit()->FiringForce()))
-			maxRange = max(maxRange, weapon.GetOutfit()->Range());
+	for(const Hardpoint &hardpoint : ship.Weapons())
+		if(hardpoint.IsReady()
+				&& !(!currentTarget && hardpoint.IsHoming() && hardpoint.GetOutfit()->Ammo())
+				&& !(!secondary && hardpoint.GetOutfit()->Icon())
+				&& !(beFrugal && hardpoint.GetOutfit()->Ammo())
+				&& !(isWaitingToJump && hardpoint.GetOutfit()->FiringForce()))
+			maxRange = max(maxRange, hardpoint.GetOutfit()->Range());
```

Comment:
Why exactly the changes here?

Reply:
Oh, this 'thing' has been shrinked a lot (due to the new members) but I felt that having the same name of the -iterated thing- every time a function return a collection is more readable, so I changed every occurrence to 'hardpoint' (not 'weapon' because that helps to highlight some element can also be empty).

Reply:
ok, fine for me